subsys/disk/disk_access_sdhc.h

```diff
@@ -0,0 +1,746 @@
+	0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+	0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+	0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+	0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+};
+
+BUILD_ASSERT(sizeof(sdhc_ones) % SDHC_CSD_SIZE == 0);
+BUILD_ASSERT(SDMMC_DEFAULT_BLOCK_SIZE % sizeof(sdhc_ones) == 0);
+
+/* Maps R1 response flags to error codes */
+static const struct sdhc_flag_map sdhc_r1_flags[] = {
+	{SDHC_R1_PARAMETER, EFAULT},	   {SDHC_R1_ADDRESS, EFAULT},
+	{SDHC_R1_ILLEGAL_COMMAND, EINVAL}, {SDHC_R1_COM_CRC, EILSEQ},
+	{SDHC_R1_ERASE_SEQ, EIO},	  {SDHC_R1_ERASE_RESET, EIO},
+	{SDHC_R1_IDLE, ECONNRESET},	{0, 0},
+};
+
+/* Maps disk status flags to error codes */
+static const struct sdhc_flag_map sdhc_disk_status_flags[] = {
+	{DISK_STATUS_UNINIT, ENODEV},
+	{DISK_STATUS_NOMEDIA, ENOENT},
+	{DISK_STATUS_WR_PROTECT, EROFS},
+	{0, 0},
+};
+
+/* Maps data block flags to error codes */
+static const struct sdhc_flag_map sdhc_data_response_flags[] = {
+	{SDHC_RESPONSE_WRITE_ERR, EIO},
+	{SDHC_RESPONSE_CRC_ERR, EILSEQ},
+	{SDHC_RESPONSE_ACCEPTED, 0},
+	/* Unrecognised value */
+	{0, EPROTO},
+};
+
+/* Returns true if an error code is retryable at the disk layer */
+static inline bool sdhc_is_retryable(int err)
+{
+	switch (err) {
+	case 0:
+		return false;
+	case -EILSEQ:
+	case -EIO:
+	case -ETIMEDOUT:
+		return true;
+	default:
+		return false;
+	}
+}
+
+/* Maps a flag based error code into a Zephyr errno */
+static inline int sdhc_map_flags(const struct sdhc_flag_map *map, int flags)
+{
+	if (flags < 0) {
+		return flags;
+	}
+
+	for (; map->mask != 0U; map++) {
+		if ((flags & map->mask) == map->mask) {
+			return -map->err;
+		}
+	}
+
+	return -map->err;
+}
+
+/* Converts disk status into an error code */
+static inline int sdhc_map_disk_status(int status)
+{
+	return sdhc_map_flags(sdhc_disk_status_flags, status);
+}
+
+/* Converts the R1 response flags into an error code */
+static inline int sdhc_map_r1_status(int status)
+{
+	return sdhc_map_flags(sdhc_r1_flags, status);
+}
+
+/* Converts an eary stage idle mode R1 code into an error code */
+static inline int sdhc_map_r1_idle_status(int status)
+{
+	if (status < 0) {
+		return status;
+	}
+
+	if (status == SDHC_R1_IDLE) {
+		return 0;
+	}
+
+	return sdhc_map_r1_status(status);
+}
+
+/* Converts the data block response flags into an error code */
+static inline int sdhc_map_data_status(int status)
+{
+	return sdhc_map_flags(sdhc_data_response_flags, status);
+}
+
+/* Initialises a retry helper */
+static inline void sdhc_retry_init(struct sdhc_retry *retry, u32_t timeout,
+			    u16_t sleep)
+{
+	retry->end = k_uptime_get_32() + timeout;
+	retry->tries = 0;
+	retry->sleep = sleep;
+}
+
+/* Called at the end of a retry loop.  Returns if the minimum try
+ * count and timeout has passed.  Delays/yields on retry.
+ */
+static inline bool sdhc_retry_ok(struct sdhc_retry *retry)
```

Comment:
Why did you move it to `disk_access_sdhc.h`, if it is only used in `disk_access_spi_sdhc.c`? The same questions applies to other functions in `disk_access_sdhc.h` which exist in one of the `disk_access_*.c` files?

Reply:
@junyangnxp please respond

Reply:
Actually I moved all the functions which are not associated to SPI from disk_access_spi_sdhc.c to disk_access_sdhc.h. Although some functions only are used by disk_access_spi_sdhc.c now, they are potential to be used by future other vender's SDHC drivers.